tests/ceph_volume/test_ceph_volume_wrong_db_size_compute.py

```diff
@@ -0,0 +1,108 @@
+
+
+def _add_node_volumes(nodes, config, size, count):
+    """Create volume in bulk and attach to node
+    Args:
+        nodes (ceph): nodes on volumes to create
+        config (dict): ceph config
+        size (str): size of volume
+        count (int): number of volume
+    """
+    # Get cloud creadentials
+    cloud_type = config.get("cloud-type")
+    cloud_configs = get_cloud_credentials(cloud_type)
+    cloud = CloudProvider(cloud_type, **cloud_configs)
+
+    nodes = nodes if type(nodes) in (list, tuple) else [nodes]
+    for node in nodes:
+        for i in range(count):
+            volname = f"{node.ceph_nodename}-vol-{size}-{i}"
+            Volume(volname, cloud).create(size=size)
+            Node(node.ceph_nodename, cloud).attach_volume(volname)
+    return True
+
+
+def _identify_disks(node):
+    """
+    Identifies the disks associated with the node
+    Args:
+        node (ceph): Node in which the cmd is executed
+
+    Returns (list): List of available disks
+    """
+
+    disks = []
+    conf = {"format": "json"}
+    available_disks = json.loads(CephAdm(node).ceph.orch.device.ls(**conf))
+    for key in available_disks:
+        if key.get("devices"):
+            for dev in key.get("devices"):
+                disks.append(dev.get("path"))
+    return list(set(disks))
+
+
+def run(ceph_cluster, **kw):
+    """Wrong block_db_size computed when adding OSD"""
+
+    # Get configs
+    get_configs()
+    config = kw.get("config")
+
+    # Get the installer and OSD nodes
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    osd_nodes = ceph_cluster.get_nodes(role="osd")
+
+    # Get specs from config
+    device_spec = config.get("specs", {})
+
+    # Add RHOS-d volume for data device
+    size = device_spec.get("block_db_size", {}).replace("G", "")
+    _add_node_volumes(nodes=osd_nodes, config=config, size=size, count=6)
+
+    # Identify available devices on node
+    disks = _identify_disks(installer)
+
+    # Get required host from config specs
+    nodes = config.get("specs", {}).get("placement", {}).get("hosts")
+    host = [ceph_cluster.get_nodes()[int(node[-1])].hostname for node in nodes]
+
+    # Create spec file specific to hostname, and devices
+    specs = config.get("specs", {})
+    specs["data_devices"]["paths"] = disks[3:]
+    specs["db_devices"]["paths"] = disks[:2]
+    specs["placement"]["hosts"] = host
+    specs["block_db_size"] = size
+
+    # Refresh ceph orch devices
+    c = {"refresh": True}
+    devices = CephAdm(installer).ceph.orch.device.ls(**c)
+    if not devices:
+        raise OperationFailedError("Devices are not re-freshed")
+
+    # Create a spec file after updating devices
+    file = create_yaml_config(installer, specs)
+
+    # Create OSDs with spec file
+    c = {"pos_args": [], "input": file}
+    CephAdm(nodes=installer, mount=file).ceph.orch.apply(**c)
+
+    # Sleep for 60s for OSDs to be generate
+    time.sleep(60)
```

Comment:
why constant sleep ? can we use waiter for OSD check ?

Reply:
I'll explain, so when we apply spec, generating OSD ids takes few seconds, if we run without sleep, it gives index out of range. 

Why waiter won't work here is, I have wait_for_osd_daemon_state to check if OSDs are up and running there we are using waiter again. 

So using time.sleep will look a good approach. Let me know if you still think differently.